lib/internal/fs/watchers.js

```diff
@@ -163,7 +162,9 @@ FSWatcher.prototype[kFSWatchStart] = function(filename,
   if (this._handle === null) {  // closed
     return;
   }
-  assert(this._handle instanceof FSEvent, 'handle must be a FSEvent');
```

Comment:
I prefer to keep it as is. Use of assertions to ensure that requirements are met is a common pattern used by many languages. Also the `ERR_UNEXPECTED_INSTANCE` custom error does not make sense in my opinion. It's an error that should not exist.

That said I won't block this if others are ok with removing assertions from the code base.

Reply:
> prefer to keep it as is. Use of assertions to ensure that requirements are met is a common pattern used by many languages.

I certainly am not excited about adding another error code--wish we had vastly fewer codes--so could leave the assert() part alone. This PR would then just add the test case and remove the expectInternalAssertion() from common.